folsom/src/main/java/com/spotify/folsom/MemcacheClient.java

```diff
@@ -44,6 +54,18 @@
    */
   CompletionStage<MemcacheStatus> set(String key, V value, int ttl, long cas);
 
+  /**
+   * Compare and set a key in memcache to the provided value, with the specified TTL
+   *
+   * @param key The key, must not be null
+   * @param value The value, must not be null
+   * @param ttl The TTL in seconds
+   * @param cas The CAS value, must match the value on the server for the set to go through
+   * @param flags Memcached flags
+   * @return A future representing completion of the request.
+   */
+  CompletionStage<MemcacheStatus> set(String key, V value, int ttl, long cas, int flags);
```

Comment:
Maybe we need to make all new interface methods `default` to avoid breaking compatibility if external code is implementing these interfaces.

In this case, the default method could just call the other set method